Build failure: python311Packages.keyutils

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `'github:NixOS/nixpkgs/8efd5d1e283604f75a808a20e6cde0ef313d07d4'#python311Packages.keyutils`, or anything depending on it (my failure specifically was triggered by udiskie).

### Build log

```
❯ nix build 'github:NixOS/nixpkgs/8efd5d1e283604f75a808a20e6cde0ef313d07d4'#python311Packages.keyutils
error: builder for '/nix/store/lhcwvc1d9ihdnxn455wb5q9mvhg5c8j1-python3.11-keyutils-0.6.drv' failed with exit code 1;
       last 10 log lines:
       > building 'keyutils._keyutils' extension
       > creating build/temp.linux-aarch64-cpython-311
       > creating build/temp.linux-aarch64-cpython-311/keyutils
       > gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/nix/store/45307rq4fz0cy70f1s07wpy82rd8shy3-libxcrypt-4.4.36/include -fPIC -I/nix/store/nxxghhhia9x1skpqfak4z407yy0mbnnd-python3-3.11.5/include/python3.11 -c keyutils/_keyutils.c -o build/temp.linux-aarch64-cpython-311/keyutils/_keyutils.o
       > keyutils/_keyutils.c:196:12: fatal error: longintrepr.h: No such file or directory
       >   196 |   #include "longintrepr.h"
       >       |            ^~~~~~~~~~~~~~~
       > compilation terminated.
       > error: command '/nix/store/pnq8ll8ayvlkw6p0xh9fmqn94v4lfi78-gcc-wrapper-12.3.0/bin/gcc' failed with exit code 1
       > /nix/store/j4w7yj8c4z7zg5y6bhhypnl456dwmrc5-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/lhcwvc1d9ihdnxn455wb5q9mvhg5c8j1-python3.11-keyutils-0.6.drv'.
```

Looks like the exact same issue that is being reported in Hydra, on the same hash: https://hydra.nixos.org/build/239103890/nixlog/1

### Additional context

Seems similar to https://github.com/NixOS/nixpkgs/issues/239816. https://github.com/sassoftware/python-keyutils/blob/5e6d2af113f6f6d2f0278ba700be23c79a08ef4b/keyutils/_keyutils.c#L1 reports that it was generated with cython 0.29.3.

Unfortunately https://github.com/sassoftware/python-keyutils also seems to be archived and a cursory check does not seem to show a fork floating around.

### Notify maintainers

@primeos

### Metadata

```console
❯ nix run nixpkgs#nix-info -- -m
 - system: `"aarch64-linux"`
 - host os: `Linux 6.1.57, NixOS, 23.11 (Tapir), 23.11.20231013.a9f8820`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - nixpkgs: `/nix/store/sv86xgs5nm17i5b0kv14151f3yhv57cx-source`
```

--- a/pkgs/development/python-modules/keyutils/default.nix
+++ b/pkgs/development/python-modules/keyutils/default.nix
@@ -1,4 +1,10 @@
-{ lib, buildPythonPackage, fetchFromGitHub, keyutils, pytest-runner, pytest }:
+{ lib
+, buildPythonPackage
+, cython
+, fetchFromGitHub
+, keyutils
+, pytestCheckHook
+}:
 
 buildPythonPackage rec {
   pname = "keyutils";
@@ -16,8 +22,19 @@ buildPythonPackage rec {
     substituteInPlace setup.py --replace '"pytest-runner"' ""
   '';
 
+  preBuild = ''
+    cython keyutils/_keyutils.pyx
+  '';
+
+  preCheck = ''
+    rm -rf keyutils
+  '';
+
   buildInputs = [ keyutils ];
-  nativeCheckInputs = [ pytest pytest-runner ];
+  nativeBuildInputs = [ cython ];
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
 
   meta = {
     description = "A set of python bindings for keyutils";

